taichi/ir/statements.h

```diff
@@ -240,6 +244,37 @@ class LoopIndexStmt : public Stmt {
   TI_DEFINE_ACCEPT_AND_CLONE
 };
 
+class BlockCornerIndexStmt : public Stmt {
+ public:
+  Stmt *loop;
+  int index;
+
+  BlockCornerIndexStmt(Stmt *loop, int index) : loop(loop), index(index) {
+    TI_STMT_REG_FIELDS;
+  }
+
+  bool has_global_side_effect() const override {
+    return false;
+  }
+
+  TI_STMT_DEF_FIELDS(ret_type, loop, index);
+  TI_DEFINE_ACCEPT_AND_CLONE
+};
+
+class BlockDimStmt : public Stmt {

```

Comment:
Is this the same as `current_struct_for->snode->num_active_indices`?

Reply:
This is actually `offload->block_dim`. However, since now we have this value at compile-time, I'm directly emitting a `ConstStmt` there.

Reply:
So the reason for not just using `ConstStmt(offload->block_dim)` is that we want to support the case we don't know the value at compile-time or at some stage of compilation?

Reply:
Yeah, for example, in the future a user may want to construct something like `print(ti.block_dim())` in a Taichi kernel.

Reply:
Sorry about the confusion. I added this statement and then realized the systematic and performant solution is to move `block_dim` evaluation during `offload`. After that, this statement is useless for now, but I kept it here since we may need this in the future.